sqlglot/expressions.py

```diff
@@ -637,7 +637,7 @@ def subquery(self, alias=None, copy=True):
         instance = _maybe_copy(self, copy)
         return Subquery(
             this=instance,
-            alias=to_identifier(alias),
+            alias=TableAlias(this=to_identifier(alias)),
```

Comment:
can you do alias_(alias, table=True)

Reply:
Mm I don't think so - we need to return a Subquery instance.